### ISSUE TITLE
Error in PDB Analyzer

### ISSUE THREAD

**Issue by danthespal** [NationalSecurityAgency/ghidra#1030]
**Describe the bug**
A clear and concise description of the bug.

**To Reproduce**
1. Open Ghidra
2. In the analyze proccess the error ocur

**Expected behavior**
Should load PDB file 

**Screenshots**

**Attachments**
```java
Problem parsing or applying PDB information: Index -1 out of bounds for length 0
java.io.IOException: Problem parsing or applying PDB information: Index -1 out of bounds for length 0
	at ghidra.app.util.bin.format.pdb.PdbParser.applyTo(PdbParser.java:409)
	at ghidra.app.plugin.core.analysis.PdbAnalyzer.parsePdb(PdbAnalyzer.java:150)
	at ghidra.app.plugin.core.analysis.PdbAnalyzer.added(PdbAnalyzer.java:80)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:185)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:685)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:785)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:664)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:629)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:62)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:101)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:458)
	at ghidra.app.util.bin.format.pdb.DefaultCompositeMember.addUnionMember(DefaultCompositeMember.java:925)
	at ghidra.app.util.bin.format.pdb.DefaultCompositeMember.addMember(DefaultCompositeMember.java:1049)
	at ghidra.app.util.bin.format.pdb.DefaultCompositeMember.addMember(DefaultCompositeMember.java:501)
	at ghidra.app.util.bin.format.pdb.DefaultCompositeMember.applyDataTypeMembers(DefaultCompositeMember.java:1138)
	at ghidra.app.util.bin.format.pdb.ApplyDataTypes.buildDataTypes(ApplyDataTypes.java:125)
	at ghidra.app.util.bin.format.pdb.PdbParser.completeDefferedTypeParsing(PdbParser.java:285)
	at ghidra.app.util.bin.format.pdb.PdbParser.applyTo(PdbParser.java:366)
	... 12 more
```
---------------------------------------------------
Build Date: 2019-Sep-20 1839 EEST
Ghidra Version: 9.1
Java Home: C:\Program Files\Java\jdk-11.0.4
JVM Version: Oracle Corporation 11.0.4
OS: Windows 10 10.0 amd64
Workstation: DESKTOP-0G9D2VD`

**Environment (please complete the following information):**
 - OS: [e.g. macOS 10.14.2]
 - Java Version: [e.g. 11.0]
 - Ghidra Version: [e.g. 9.0]

**Additional context**

**Comment by aerosoul94** [comment]
I was able to replicate this issue by generating a PDB with the union below.
```
union TestUnion {
    struct {
        int :32;
        int :11;
        int field1:1;
        int field2:4;
    };
};
```

Problem lies here: https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/DefaultCompositeMember.java#L917

Because the first field's offset is not at 0, it never fills in `unionMemberList`, leaving it empty, and so it ends up calling `unionMemberList.get(unionMemberList.size() - 1)` passing -1 as the index.

**Comment by ghizard** [comment]
@fenugrec The problem that you have found is different from the original problem.  Your issues was fixed with 953f59317ae535cefed0949794972f8878d137b1.  This fix is available starting with Ghidra 11.1.

@danthespal @aerosoul94 I've was able to reproduce the problem, given @aerosoul94's example code.  This will likely need the attention of @ghidra1.

**Comment by ghizard** [comment]
Regarding the initial issue, I tried a naive approach of testing for `unionMemberList.isEmpty()`, which I didn't expect would be correct, as the test for `member.memberOffset == 0` could be gathering multiple bit-fields.
https://github.com/NationalSecurityAgency/ghidra/blob/86632adc77e5991a1c9689cbc7d1aabf12c45257/Ghidra/Features/PDB/src/main/java/ghidra/app/util/bin/format/pdb/DefaultCompositeMember.java#L929-L933
This gets us past the problem (it does not fail), but it creates issues in other data types.
I'm deferring this to @ghidra1
